File: LC#217-Contains_Duplicate/contains_duplicate.py
# 217. Contains Duplicate
# Difficulty: Easy
# Topics: Array, Hash Table, Sorting
# https://leetcode.com/problems/contains-duplicate/
from typing import List

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        uniqueNums = set()
        for num in nums:
            if num in uniqueNums:
                return True
            uniqueNums.add(num)
        return False
            
solution_instance = Solution()

# Test cases
case_1 = [1, 2, 3, 1]
case_2 = [1, 2, 3, 4]
case_3 = [1, 1, 1, 3, 3, 4, 3, 2, 4, 2]

assert solution_instance.containsDuplicate(case_1)
assert not solution_instance.containsDuplicate(case_2)
assert solution_instance.containsDuplicate(case_3)

print("All tests passed successfully!")

File: LC#217-Contains_Duplicate/contains_duplicate.rs
use std::collections::HashSet;
fn main() {
    fn contains_duplicate(nums: Vec<i32>) -> bool {
        let mut unique_nums: HashSet<i32> = HashSet::new();
        for num in nums {
            if unique_nums.contains(&num) {
                return true
            }
            unique_nums.insert(num);
        }
        false
    }
    
    // Test cases
    let case_1 = vec!(1, 2, 3, 1);
    let case_2 = vec!(1, 2, 3, 4);
    let case_3 = vec!(1, 1, 1, 3, 3, 4, 3, 2, 4, 2);
    
    assert_eq!(contains_duplicate(case_1), true);
    assert_eq!(contains_duplicate(case_2), false);
    assert_eq!(contains_duplicate(case_3), true);
    
    println!("All tests passed successfully!");
}